etl/grapher_helpers.py

```diff
@@ -286,3 +287,143 @@ def join_sources(sources: List[catalog.meta.Source]) -> catalog.meta.Source:
+        )
+        if attribute == "description":
+            if metadata.description is not None:
+                # Add the dataset description as if it was a source's description.
+                values = [metadata.description] + values
+
+            # Descriptions are usually long, so it is better so put together descriptions from different sources in
+            # separate lines.
+            combined_value = "\n".join(values)
+        elif attribute == "date_accessed":
+            # For dates simply take the one from the first source.
+            combined_value = values[0]
+        else:
+            # For any other attribute, values from different sources can be in the same line, separated by ;.
+            combined_value = " ; ".join(values)
+
+        setattr(default_source, attribute, combined_value)
+
+    # Remove other sources and keep only the default one.
+    metadata.sources = [default_source]
+
+    return metadata
+
+
+def adapt_dataset_metadata_for_grapher(
+    metadata: catalog.DatasetMeta,
+) -> catalog.DatasetMeta:
+    """Adapt metadata of a garden dataset to be used in a grapher step.
+
+    Parameters
+    ----------
+    metadata : catalog.DatasetMeta
+        Dataset metadata.
+
+    Returns
+    -------
+    metadata : catalog.DatasetMeta
+        Adapted dataset metadata, ready to be inserted into grapher.
+
+    """
+    # Combine metadata sources into one.
+    metadata = combine_metadata_sources(metadata)
+
+    # Add institution and year to dataset short name (the name that will be used in grapher database).
+    short_name_ending = "__" + underscore(f"{metadata.namespace}_{metadata.version}")
+    if not metadata.short_name.endswith(short_name_ending):
+        metadata.short_name = metadata.short_name + short_name_ending
+    # Empty dataset description (otherwise it will appear in `Internal notes` in the admin UI).
+    metadata.description = ""
+
+    return metadata
+
+
+def adapt_table_for_grapher(
+    table: catalog.Table, country_col: str = "country", year_col: str = "year"
+) -> catalog.Table:
+    """Adapt table (from a garden dataset) to be used in a grapher step.
+
+    Parameters
+    ----------
+    table : catalog.Table
+        Table from garden dataset.
+    country_col : str
+        Name of country column in table.
+    year_col : str
+        Name of year column in table.
+
+    Returns
+    -------
+    table : catalog.Table
+        Adapted table, ready to be inserted into grapher.
+
+    """
+    table = deepcopy(table)
+    # Grapher needs a column entity id, that is constructed based on the unique entity names in the database.
+    table["entity_id"] = country_to_entity_id(table[country_col], create_entities=True)
+    table = table.drop(columns=[country_col]).rename(columns={year_col: "year"})
+    table = table.set_index(["entity_id", "year"])
+
+    # Ensure the default source of each column includes the description of the table (since that is the description that
```

Comment:
This is already done in [grapher_import](https://github.com/owid/etl/blob/master/etl/grapher_import.py#L190), but it's gonna be better to do it explicitly with your function and raise a warning if source is missing. I'll create a PR after you merge this.